### DISCUSSION TITLE
How to Keep the order of boxes?

### DISCUSSION THREAD

**Discussion by Muhammadraafat1** [pymupdf/PyMuPDF#3529] [Looking for help] [answered]
Hi, When I applied redaction and insert_text with the replaced word and save my output.pdf, I try by mouse to select line has original word next to replaced word, I found all pages in pdf was selected!

Here is an example:
this is original text: As you see it select what I already selected by mouse
![image](https://github.com/pymupdf/PyMuPDF/assets/140750834/dec6713d-34fa-4be0-a483-75ad1fa96f9e)

Here I want to select only "To: PETER ADEL":  PETER ADEL is replacement word.
![image](https://github.com/pymupdf/PyMuPDF/assets/140750834/d5613642-ee27-4805-8ef8-61d20aed580d)

There is anything to do with that?

**Comment by JorjMcKie** [comment, marked as answer]
This lies in the nature of PDF and cannot be avoided.
Text will always be selected (and extracted) in the sequence **_as physically stored_** in the page appearance program ("stream").

In case of redactions, this means that the evacuated space of the old text will not somehow be left as a gap in that appearance stream. The respective commands displaying old text will instead be deleted.
And the new text will be appended via program-like instructions **_to the end of that program_**.
Everything **_appears_** to be in nice order, because the new text carries coordinates instructing the PDF viewer to put it where intended.
